Borrow API update & Security

joeflack4 opened the pull request:
I will be using this branch to deploy borrow web. I think it is solid enough to merge. But we can review and change things if you would like.

## Security
- Security updates via dependency version up.

## Developer updates 
The following was done to improve compatibility, especially with  ArgparseToWebform.
- Exposed ArgumentParser object as 'parser' to Borrow API by moving it to global scope of module. 
- Renamed CLI option 'xlsxfile' to 'xlsxfiles' to make consistent with the Python API.
- Altered Python API borrow() function definition. Removed unneeded *. Changed optional params to default to None or False. Strongly coupled CLI and Python API parameter interface by requiring CLI parameters passed to be exactly the same, via dynamically loading list of attribute names from the borrow() function definition.

joeflack4 committed:
Fixed security vulnerabilities

joeflack4 committed:
Developer updates
- Exposed ArgumentParser object as 'parser' to Borrow API by moving it to global scope of module. Rationale is to allow for use with ArgeparseToWebform.
- Renamed CLI option 'xlsxfile' to 'xlsxfiles' to make consistent with the Python API.
- Altered Python API borrow() function definition. Removed unneeded *. Changed optional params to default to None or False. Strongly coupled CLI and Python API parameter interface by requiring CLI parameters passed to be exactly the same, via dynamically loading list of attribute names from the borrow() function definition.

jkpr commented on pmix/borrow.py:
```diff
@@ -170,97 +240,30 @@ def merge_translation_file(merge: List[str], translation_dict: TranslationDict,
-           merge_all: List[str], add: List[str], ignore: List[str],
-           no_diverse: bool = False, diverse: str = None, carry: bool = False,
-           outfile: str = None, outdir: str = None):
+def borrow(
```
Leave `*` here. The borrow function has too many parameters that knowing the order doesn't help. I want all arguments to be keyword only.

joeflack4 replied:
@jkpr Ahh, ok. That is what the `*` was for. I asked you if you knew what this was on Slack, but you must have missed it. And I couldn't find the answer by googling. Will add.

jkpr commented on pmix/borrow.py:
```diff
@@ -170,97 +240,30 @@ def merge_translation_file(merge: List[str], translation_dict: TranslationDict,
-        outfile=args.outfile,
-        outdir=args.outdir,
-    )
+    borrow_vars: List[str] = borrow.__code__.co_varnames
```
Python has a better way: `vars(args)`. This gives you the parsed args as a dictionary. Would much prefer to go this route rather than what you have done.

joeflack4 replied:
@jkpr Hah, yes I agree. I forgot about that.

jkpr commented on pmix/borrow.py:
```diff
@@ -31,12 +31,82 @@
 
 
-def create_translation_dict(xlsxfile: List[str], correct: List[str]) \
+parser = argparse.ArgumentParser(
```
Let's put this `ArgumentParser` in a function that returns the ArgumentParser. I prefer not to have the parser as a global variable in the module.

jkpr reviewed (changes requested):
I have made some comments on your changes. Please review and I look forward to your feedback ☄️

joeflack4 commented:
@jkpr Thanks for the feedback! I'll prioritize this after the Ethiopia forms, but eager to get back to this.

joeflack4 committed:
Developer updates
- Refactored borrow_cli() function to be much cleaner.
- Re-added leading * to borrow() function definition to require keyword args.
- Removed parser obj from global module variable in favor of cli_parser() constructor.
- Updated .gitignore

joeflack4 commented:
@jkpr Requested changes have been made!

jkpr commented on pmix/borrow.py:
```diff
@@ -170,97 +249,30 @@ def merge_translation_file(merge: List[str], translation_dict: TranslationDict,
+        attr: getattr(args, attr)
+        for attr in vars(borrow)
+    }
+    borrow(**kwargs)
```
Why not simply `borrow(**vars(args))` ?

joeflack4 replied:
@jkpr Didn't think of that during the update! Makes even more sense ;3. I'll send a new commit
